tools/packaging/kata-deploy/scripts/kata-deploy.sh

```diff
@@ -268,7 +268,8 @@ function main() {
 		containerd_conf_file_backup="${containerd_conf_file}.bak"
 	else
 		# runtime == containerd
-		if [ ! -f "$containerd_conf_file" ]; then
```

Comment:
I'm not expert on kata-deploy so caution with my suggestion: I think this whole if-then-else block needs a refactoring because it does some initialization that are specific to given actions and the actions are called just later.

For example, if `runtime` is `k3s` and action is `cleanup` then copying `containerd_conf_file` to `containerd_conf_tmpl_file` still make sense?

In other words, you fixed one case but others might be broken because the initialization is wrong there.

Reply:
right, although in that case it's less urgent as it's not harmful.
Thing is, the difficulty of testing changes on kata-deploy makes it discouraging to touch.
I'd suggest to have currently it as is and backport to stable to unblock the release